rules/merge_inputs.smk

```diff
@@ -13,60 +13,50 @@ def _parse_config_input(input):
       in which case it must not include the wildcard substring.
 
     Returns a dictionary with optional keys:
-    - metadata:string - the relative path to the metadata file. If the original data was remote then this represents
-      the output of a rule which downloads the file
-    - metadata_location:string - the URI for the remote file if applicable else `None`
-    - sequences:function. Takes in wildcards and returns the relative path to the sequences FASTA for the provided
+    - metadata:string - path or url to the metadata file.
+    - sequences:function. Takes in wildcards and returns path or url to the sequences FASTA for the provided
       segment wildcard, or returns `None` if this input doesn't define sequences for the provided segment.
-    - sequences_location:function. Takes in wildcards and returns the URI for the remote file, or `None`, where applicable.
 
     Raises InvalidConfigError
     """
-    name = input['name']
-    lambda_none = lambda w: None
-
-    info = {'metadata': None, 'metadata_location': None, 'sequences': lambda_none, 'sequences_location': lambda_none}
-
-    def _source(uri, *,  s3, local):
-        if uri.startswith('s3://'):
-            return s3
-        elif uri.lower().startswith('http://') or uri.lower().startswith('https://'):
-            raise InvalidConfigError("Workflow cannot yet handle HTTP[S] inputs")
-        return local
-
-    if location:=input.get('metadata', False):
-        info['metadata'] = _source(location,  s3=f"data/{name}/metadata.tsv", local=location)
-        info['metadata_location'] = _source(location,  s3=location, local=None)
+    info = {
+        "name": input["name"],
+        "metadata": path_or_url(input["metadata"]) if input.get("metadata") else None,
+        "sequences": None,
+    }
 
     if location:=input.get('sequences', False):
         if isinstance(location, dict):
-            info['sequences'] = lambda w: _source(location[w.segment],  s3=f"data/{name}/sequences_{w.segment}.fasta", local=location[w.segment]) \
-                if w.segment in location \
-                else None
-            info['sequences_location'] = lambda w: _source(location[w.segment], s3=location[w.segment], local=None) \
+            info['sequences'] = lambda w: path_or_url(location[w.segment]) \
                 if w.segment in location \
                 else None
         elif isinstance(location, str):
-            info['sequences'] = _source(location, s3=lambda w: f"data/{name}/sequences_{w.segment}.fasta", local=lambda w: location.format(segment=w.segment))
-            info['sequences_location'] = _source(location,  s3=lambda w: location.format(segment=w.segment), local=lambda_none)
+            info['sequences'] = lambda w: path_or_url(location)
```

Comment:
The `location` string can have a `{segment}` wildcard placeholder in it ([docs](https://github.com/nextstrain/avian-flu?tab=readme-ov-file#running-the-segment-level-builds-using-ncbi-data)). Does this work in `path_or_url`? (I can't remember if that ultimately returns a snakemake input/output function which handles the wildcards)

Previously this was filled in via `lambda w: location.format(segment=w.segment)`

Reply:
Yeah, it was a nice discovery that the Snakemake storage plugins _just work_ with wildcards.